dotcom-rendering/src/web/components/MultiImageBlockComponent.tsx

```diff
@@ -89,185 +89,249 @@ const GridOfFour = ({ children }: { children: React.ReactNode }) => (
+	format: ArticleFormat;
+	caption?: string;
+}) => (
+	<div
+		css={css`
+			margin-top: 12px;
+			margin-bottom: 12px;
+
+			img {
+				object-fit: cover;
+				width: 100%;
+			}
+		`}
+	>
+		<OneAboveTwoGrid>
+			<GridItem area="first">
+				<ImageComponent
+					element={images[0]}
+					format={format}
+					hideCaption={true}
+					role={images[0].role}
+				/>
+			</GridItem>
+			<GridItem area="second">
+				<ImageComponent
+					element={images[1]}
+					format={format}
+					hideCaption={true}
+					role={images[1].role}
+				/>
+			</GridItem>
+			<GridItem area="third">
+				<ImageComponent
+					element={images[2]}
+					format={format}
+					hideCaption={true}
+					role={images[2].role}
+				/>
+			</GridItem>
+		</OneAboveTwoGrid>
+		{!!caption && (
+			<Caption
+				captionText={caption}
+				format={format}
+				shouldLimitWidth={false}
+			/>
+		)}
+	</div>
+);
+
+const FourImage = ({
+	images,
+	format,
+	caption,
+}: {
+	images: [
+		ImageBlockElement,
+		ImageBlockElement,
+		ImageBlockElement,
+		ImageBlockElement,
+	];
+	format: ArticleFormat;
+	caption?: string;
+}) => (
+	<div
+		css={css`
+			margin-top: 12px;
+			margin-bottom: 12px;
+			img {
+				object-fit: cover;
+				width: 100%;
+			}
+		`}
+	>
+		<GridOfFour>
+			<GridItem area="first">
+				<ImageComponent
+					element={images[0]}
+					format={format}
+					hideCaption={true}
+					role={images[0].role}
+				/>
+			</GridItem>
+			<GridItem area="second">
+				<ImageComponent
+					element={images[1]}
+					format={format}
+					hideCaption={true}
+					role={images[1].role}
+				/>
+			</GridItem>
+			<GridItem area="third">
+				<ImageComponent
+					element={images[2]}
+					format={format}
+					hideCaption={true}
+					role={images[2].role}
+				/>
+			</GridItem>
+			<GridItem area="forth">
+				<ImageComponent
+					element={images[3]}
+					format={format}
+					hideCaption={true}
+					role={images[3].role}
+				/>
+			</GridItem>
+		</GridOfFour>
+		{!!caption && (
+			<Caption
+				captionText={caption}
+				format={format}
+				shouldLimitWidth={false}
+			/>
+		)}
+	</div>
+);
+
 export const MultiImageBlockComponent = ({
 	images,
 	format,
 	caption,
 }: Props) => {
-	const imageCount = images.length;
+	const [one, two, three, four] = images;
 
-	switch (imageCount) {
-		case 1:
-			return (
-				<div
-					css={css`
-						margin-top: 12px;
-						margin-bottom: 12px;
-						img {
-							object-fit: cover;
-							width: 100%;
-						}
-					`}
-				>
-					<ImageComponent
-						format={format}
-						element={images[0]}
-						hideCaption={true}
-						role={images[0].role}
-					/>
-					{!!caption && (
-						<Caption
-							format={format}
-							captionText={caption}
-							shouldLimitWidth={false}
-						/>
-					)}
-				</div>
-			);
-		case 2:
-			return (
-				<div
-					css={css`
-						margin-top: 12px;
-						margin-bottom: 12px;
+	if (one && two && three && four) {
```

Comment:
tempting to abstract the <div> wrapper as I think that styling is repeated across all four components? but not a big deal (and maybe not one for this PR anyway?)

either way, this reads much better to me, nice! 👍